Project.toml

```diff
@@ -5,6 +5,7 @@ version = "0.8.0"
 [deps]
 ChainRulesCore = "d360d2e6-b24c-11e9-a2a3-2a2ae2dbcce4"
 Compat = "34da2185-b29b-5c13-b0c7-acf172513d20"
+DataStructures = "864edb3b-99cc-5e75-8d2d-829cb0a9cfe8"
```

Comment:
we are trying to stay lean, so we should avoid adding this dependency if possible

Reply:
Sure! I will try to avoid it.